src/Relude/Foldable/Fold.hs

```diff
@@ -76,6 +77,13 @@ foldMapA f = foldr step (pure mempty)
     step a mb = mappend <$> f a <*> mb
 {-# INLINE foldMapA #-}
 
+{- | Alternative version of @asum@
+>>> asumMap (\x -> if x > 2 then Just x else Nothing) [1..3]
+Just 3
+-}
+asumMap :: (Foldable f, Alternative m) => (a -> m b) -> f a -> m b
+asumMap f = getAlt . foldMap (\a -> Alt (f a))
+
```

Comment:
Could you add `{-# INLINE asumMap #-}` pragma after this function? I guess it might be useful. Don't think it will harm.